### ISSUE TITLE
Add Sentry static API

### ISSUE THREAD

**Issue by marandaneto** [getsentry/sentry-dart#101]
We'd like to add the Sentry static API to Dart:

https://develop.sentry.dev/sdk/unified-api/#static-api

the idea is one will be able to `Sentry.captureException(x)`, `Sentry.captureMessage(x)`, `Sentry.captureEvent(x)`, `Sentry.init(options -> {})`
Today the SDK is initialized via `SentryClient` which has no `Hub` and `Scope`.

https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/Sentry.java#L104-L111

we don't need to care for `globalHubMode` that we have on Java.
Also, we don't need a `ThreadLocal` on Dart because the language is already single-threaded (is that the right term?).

This issue could have sub-issues like:

* Adding SentryOptions to configure the SDK (only the bare min. now)
- https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/SentryOptions.java
- https://docs.sentry.io/platforms/java/usage/#configuration

* Adding a diagnostic logger (so we can log the SDK ops)
- https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/SentryOptions.java#L74-L75
- https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/SystemOutLogger.java

* Adding the Hub
- https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/Hub.java
- https://develop.sentry.dev/sdk/unified-api/#hub

* Adding the Scope class (we can start with the bare min. as well)
- https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/Scope.java
- https://develop.sentry.dev/sdk/unified-api/#scope

* Adding a Breadcrumb ring buffer (it doesn't need to be thread-safe because of Dart)
- https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/Scope.java#L385-L393
- https://develop.sentry.dev/sdk/event-payloads/breadcrumbs/

So TLDR:

The `Sentry.init` creates a `Hub`, the `Hub` creates and maintains a `SentryClient` and a `Scope`.
See the workflow: https://develop.sentry.dev/sdk/unified-api/#simplified-visualization

We could have a feature branch for this issue as we'd need to collaborate together, and each item of this task would depend on each other.

**Comment by marandaneto** [comment]
Scope and breadcrumbs ring buffer is done here https://github.com/getsentry/sentry-dart/pull/109

**Comment by rxlabz** [comment]
First part (init, captureEvent, captureException) here #108
